model/canvas.cc

```diff
@@ -503,7 +503,22 @@ namespace minsky
            });
 	}      
     itemVector.clear();  
-  }  
+  }
+  
+  void Canvas::showPlotsOnTab()
+  {
+    if (itemVector.empty())
+	{
+        model->recursiveDo
+          (&GroupItems::items, [&](const Items&,Items::const_iterator i)
+           {
+             if (auto p=(*i)->plotWidgetCast())
+               if (p->plotOnTab()==false)
+				 p->togglePlotTabDisplay();	 
```

Comment:
This is something you can virtualise as tabDisplay or toggleTabDisplay (as appropriate), which would be an empty implementation in Item, and overridden in variable, godleyIcon and plotWidget.